src/server.c

```diff
@@ -5934,19 +5948,19 @@ void dismissClientMemory(client *c) {
         void *bulk = listNodeValue(ln);
         /* Default bulk size is 16k, actually it has extra data, maybe it
          * occupies 20k according to jemalloc bin size if using jemalloc. */
-        if (bulk) dismissMemory(bulk);
+        if (bulk) dismissMemory(bulk, 0);
```

Comment:
we actually know the size of the bulk (example code below).
but maybe we don't wanna bother since it's usually big.

howeer, we can skip the loop if `c->reply_bytes` is small (in some cases there may be many nodes but all of them are small).

also, let's dismiss `c->querybuf`
and `c->argv[]` if `argv_len_sum` is big.

```suggestion
        if (bulk) dismissMemory(bulk, ((clientReplyBlock*)bulk)->size);
```

Reply:
You mean there may be some small nodes after `trimReplyUnusedTailSpace`, so it is necessary to specify size in  `dismissMemory`.

Reply:
There may be small nodes due to deferred replies, or even null nodes. I don't mind passing 0, or the size (as shown in my diff), the nodes are usually big, so we only rarely have an excessive call to malloc_size.

But more important is the bottom of my comment above, about the possibility to skip the loop, and feeling other client buffers.